src/main/kotlin/JoinDialog.kt

```diff
@@ -5,25 +5,34 @@ import tornadofx.*
 
 data class JoinDetails(val channel: String)
 
-class JoinDetailsModel : ItemViewModel<JoinDetails>() {
-    private val controller: MainController by inject()
+class JoinDialogController(private val controller: MainController) : Component() {
+    private var dialog: JoinDialog? = null
+    fun create() {
+        dialog = JoinDialog(JoinDetailsModel(this))
+        dialog?.openModal()
+    }
+    fun join(channel: String) {
+        controller.joinChannel(channel)
+        dialog?.close()
```

Comment:
Should this be in the `JoinDetailsModel` instead? Or at least go through the `JoinDetailsModel`, feels a bit weird to talk to the view directly